03_Modules/Tokens/src/module/authorizer/RealTimeAuthorizer.ts

```diff
@@ -0,0 +1,126 @@
+import { AuthorizationStatusEnumType, IdTokenInfoType, IMessageContext } from '@citrineos/base';
+import { Authorization, IdToken, IdTokenInfo } from '@citrineos/data';
+import { IAuthorizer } from '@citrineos/util';
+import {
+  AuthorizationInfoAllowed,
+  ClientInformationProps,
+  CpoTenant,
+  CpoTenantProps,
+  CredentialsService,
+  ModuleId,
+  OcpiToken,
+  PostTokenParams,
+  ServerCredentialsRoleProps,
+  TokensClientApi,
+  TokensRepository,
+  UnsuccessfulRequestException,
+} from '@citrineos/ocpi-base';
+import { BadRequestError } from 'routing-controllers';
+import { ILogObj, Logger } from 'tslog';
+import { AuthorizationInfo } from '@citrineos/ocpi-base/dist/model/AuthorizationInfo';
+import { Service } from 'typedi';
+
+@Service()
+export class RealTimeAuthorizer implements IAuthorizer {
+
+  constructor(
+    private readonly logger: Logger<ILogObj>,
+    private readonly tokensRepository: TokensRepository,
+    private readonly credentialsService: CredentialsService,
+    private readonly tokensClientApi: TokensClientApi
+  ) {
+  }
+  async authorize(authorization: Authorization, context: IMessageContext): Promise<Partial<IdTokenInfoType>> {
+    const result = new IdTokenInfo();
+    result.status = AuthorizationStatusEnumType.Invalid;
+    try {
+      const ocpiToken = await this.getOcpiToken(authorization, context);
+      const cpoTenant = await this.getCpoTenant(ocpiToken, context);
+      const params = this.buildPostTokenParams(authorization, ocpiToken, cpoTenant);
+      const authorizationInfo = await this.getPostTokenResponse(params, cpoTenant);
+      if (authorizationInfo.allowed === AuthorizationInfoAllowed.Allowed) {
+        result.status = AuthorizationStatusEnumType.Accepted;
+      }
+    } catch (e: any) {
+      this.logger.error('Error: could not authorize token -' + e.message);
+    }
+    return result;
+  }
+
+  private async getOcpiToken(authorization: Authorization, context: IMessageContext): Promise<OcpiToken> {
+    const idToken = authorization.idToken as IdToken;
+    const idTokenId = idToken.id;
+    const ocpiToken = await this.tokensRepository.getOcpiTokenByAuthorizationId(authorization.id);
+    if (!ocpiToken) {
+      throw new BadRequestError(`Could not find OCPI token with id ${idTokenId}.`);
+    }
+    return ocpiToken;
+  }
+
+  private async getCpoTenant(ocpiToken: OcpiToken, context: IMessageContext): Promise<CpoTenant> {
```

Comment:
minor, seems like there's a linter warning here -- `'context' is defined but never used. Allowed unused args must match /^_/u`

Reply:
fixed